wormhole-connect/src/utils/wallet/evm.ts

```diff
@@ -12,43 +12,73 @@ import config from 'config';
 const isChainResourceKey = (key: string): key is keyof ChainResourceMap =>
   Object.keys(config.rpcs).includes(key);
 
-const CHAINS_CONFIG = Object.entries(DEFAULT_CHAINS).map(
-  ([wagmiChainName, wagmiConfig]) => {
-    if (isChainResourceKey(wagmiChainName)) {
-      const rpc = config.rpcs[wagmiChainName];
-      if (rpc) {
-        return {
-          ...wagmiConfig,
-          rpcUrls: {
-            ...wagmiConfig.rpcUrls,
-            [wagmiChainName]: {
-              http: [rpc],
-            },
-            default: {
-              http: [rpc],
-            },
-            public: {
-              http: [rpc],
-            },
-          },
-        };
-      }
-    }
-    return wagmiConfig;
+type ChainRpcUrls = (typeof DEFAULT_CHAINS)[0]['rpcUrls']['default'];
+
+const getRpcForChain = (
+  wormholeChainName: string,
+  defaultRpc: ChainRpcUrls,
+): ChainRpcUrls =>
+  wormholeChainName in config.rpcs &&
+  isChainResourceKey(wormholeChainName) &&
+  config.rpcs[wormholeChainName] !== null &&
+  config.rpcs[wormholeChainName] !== undefined &&
+  config.rpcs[wormholeChainName] !== ''
+    ? { ...defaultRpc, http: [config.rpcs[wormholeChainName]] }
+    : defaultRpc;
+
+/**
+ * Should be used to coalesce a wagmi chain name to a wormhole chain name.
+ * This is necessary because the wormhole chain names are different from the wagmi chain names.
+ *
+ * @param name a wagmi chain name
+ * @returns a wormhole chain name
+ *
+ * NOTE: mapping could be incomplete
+ */
+const coalesceWormholeChainName = (name: string) =>
```

Comment:
you could just do `name.toLowerCase()` to avoid having to have a comprehensive list of chain names here